src/platform/forms/tests/forms.unit.spec.js

```diff
@@ -1,233 +1,298 @@
+  formConfig,
+  name,
+  required = true,
+  warning = null,
+) => {
+  validProperty(formConfig, name, 'boolean', required, warning);
+};
+
+const validStringProperty = (
+  formConfig,
+  name,
+  required = true,
+  warning = null,
+) => {
+  validProperty(formConfig, name, 'string', required, warning);
+};
+
+const validNumberProperty = (
+  formConfig,
+  name,
+  required = true,
+  warning = null,
+) => {
+  validProperty(formConfig, name, 'number', required, warning);
+};
+
+const validArrayProperty = (
+  formConfig,
+  name,
+  required = true,
+  warning = null,
+) => {
+  validProperty(formConfig, name, 'array', required, warning);
+};
+
+const validFormConfigKeys = formConfig => {
+  Object.keys(formConfig).forEach(key => {
+    const warning =
+      `${formConfig.formId} has an unknown property "${key}". ` +
+      '\nPlease check that property name is correct.' +
+      '\nIf this is a new property please update https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/blob/master/packages/documentation/src/pages/forms/config-options.mdx' +
+      ' and add a test to src/platform/forms/tests/forms.unit.spec.js for this property.\n';
+    expect(formConfigKeys).to.include(key, warning);
+  });
+};
 
-    expect(includedSchemaIdsSet.size).to.not.lessThan(
-      mappedIdsSet.size,
-      'a schema may have been removed from vets-json-schema/dist/schemas',
+const validFormId = formConfig => {
+  let formId = formConfig.formId;
+  if (Object.keys(remapFormId).includes(formId)) {
+    formId = remapFormId[formId];
+  }
+
+  validStringProperty(formConfig, 'formId');
+  const schemaFormIds = Object.keys(schemas);
+  if (missingFromVetsJsonSchema.includes(formId)) {
+    expect(schemaFormIds).to.not.include(
+      formId,
+      `${formId} is in missingFromVetsJsonSchema but has a corresponding vets-json-schema, please remove from missingFromVetsJsonSchema`,
     );
-    expect(includedSchemaIdsSet.size).to.not.greaterThan(
-      mappedIdsSet.size,
-      'a schema may have been added to vets-json-schema/dist/schemas',
+  } else {
+    expect(schemaFormIds).to.include(
+      formId,
+      `the formId "${formId}" does not match a formId property in vets-json-schema/dist/schemas, add to missingFromVetsJsonSchema until such a time`,
     );
-    expect(includedSchemaIds).to.have.same.members(
-      includedFormIds,
-      'possible missing formId property in a formConfig',
+  }
+};
+
+const validMigrations = formConfig => {
+  const { migrations } = formConfig;
+  if (migrations || formConfig.version > 0) {
+    expect(migrations.length).to.equal(
+      formConfig.version,
+      'migrations length does not match version number',
     );
-  });
+    validArrayProperty({ migrations }, 'migrations');
+    expect(
+      migrations.every(migration => typeof migration === 'function'),
+    ).to.equal(true, 'migrations contains an element that is not a function');
+  }
+};
+
+const validTitle = ({ title }) => {
```

Comment:
This might make sense to rename: `validFormTitle`. I definitely mistook it for talking about an object title, which can be a custom React component, and needn't return a string.

Reply:
Renamed